crypto/evp/pmeth_lib.c

```diff
@@ -422,6 +430,38 @@ int EVP_PKEY_CTX_set_dh_pad(EVP_PKEY_CTX *ctx, int pad)
 }
 #endif
 
+int EVP_PKEY_CTX_set_signature_md(EVP_PKEY_CTX *ctx, const EVP_MD *md)
+{
+    OSSL_PARAM sig_md_params[3];
+    size_t mdsize;
+    const char *name;
+
+    /* TODO(3.0): Remove this eventually when no more legacy */
+    if (ctx->sigprovctx == NULL)
+        return EVP_PKEY_CTX_ctrl(ctx, -1, EVP_PKEY_OP_TYPE_SIG,
+                                 EVP_PKEY_CTRL_MD, 0, (void *)(md));
+
+    if (md == NULL)
+        return 1;
+
+    mdsize = EVP_MD_size(md);
+    name = EVP_MD_name(md);
+    sig_md_params[0] = OSSL_PARAM_construct_utf8_string(
```

Comment:
> I think I'll put in a PR where this call `OSSL_PARAM_construct_utf8_string` does a `strlen(3)` of its argument if the specified length is zero.

Yes, please do.

Reply:
See #9760.  Every instance of `OSSL_PARAM_construct_utf8_string` now has a zero length.

Is it worth dropping this argument entirely?  I'm less sure about this change.

Reply:
No, because the construct function is also used to give a buffer where data should end up, by ways of `get_params` / `get_ctx_params`.  So the caller must be able to specify the buffer size.